react-app/src/components/GridPage.tsx

```diff
@@ -0,0 +1,158 @@
+import GridElement from "./UI/GridElement.tsx";
+import { GridElementModel } from "../models/GridElementModel.ts";
+import { GridInitialElementsAspects } from "../static/GridInitialElementsAspects.ts";
+
+export default function GridPage() {
+  const numberOfElements = 20;
+
+  const [cssProps, setCssProps] = useState<CSSProperties>({
+    gridAutoFlow: "dense",
+    gridTemplateColumns: "100",
+    gridAutoRows: "100",
+    gap: "10",
+  });
+
+  const [selectedAspects, setSelectedAspects] = useState<AspectState[]>(
+    GridInitialElementsAspects,
+  );
+
+  const [elements, setElements] = useState(generateElements);
+
+  const renderedElements = elements.map((el, index) => (
+    <GridElement
+      aspect={el.aspectClass}
+      color={el.color}
+      index={index}
+    ></GridElement>
+  ));
+
+  const aspectButtons = selectedAspects.map((el, index) => {
+    return (
+      <GridConfigButton
+        key={index}
+        name={el.aspectText}
+        handleClick={() => {
+          checkAspect(el.id);
+        }}
+        style={el.selected ? {} : { textDecoration: "line-through" }}
+      />
+    );
+  });
+
+  function getRandomAspect() {
+    const filteredAspects = selectedAspects.filter((e) => {
+      return e.selected == true;
+    });
+
+    if (filteredAspects.length == 0) {
+      return styles[selectedAspects[0].aspectClass];
+    }
+
+    const randomIndex = Math.floor(Math.random() * filteredAspects.length);
+    return styles[filteredAspects[randomIndex].aspectClass];
+  }
+
+  function getRandomColor() {
+    const randomIndex = Math.floor(Math.random() * colors.length);
+    return colors[randomIndex];
+  }
+
+  function generateElements() {
+    const elements: Array<GridElementModel> = [];
+
+    for (let i = 0; i < numberOfElements; i++) {
+      elements.push({
+        color: getRandomColor(),
+        aspectClass: getRandomAspect(),
+      });
+    }
+
+    return [...elements];
+  }
+
+  function setGridCss(newProps: CSSProperties) {
+    const finalProps: CSSProperties = {
+      ...cssProps,
+      ...newProps,
+    };
+
+    setCssProps(finalProps);
+  }
+
+  function setGapSizeHandler(e: React.ChangeEvent<HTMLInputElement>) {
+    setGridCss({
+      gap: e.target.value,
+    });
+  }
+
+  function setElementSizeHandler(e: React.ChangeEvent<HTMLInputElement>) {
+    setGridCss({
+      gridTemplateColumns: e.target.value,
+      gridAutoRows: e.target.value,
+    });
+  }
+
+  function checkAspect(id: number) {
+    const newAspects = [...selectedAspects];
+    newAspects.forEach((element) => {
+      if (element.id == id) {
+        element.selected = !element.selected;
+      }
+    });
+
+    setSelectedAspects(newAspects);
+  }
+
+  return (
+    <div className={styles["page-wrapper"]}>
+      <div className={styles["page-content"]}>
+        <div style={{ width: "100%", backgroundColor: "inherit" }}>
+          <div className={styles["windows-container"]}>
```

Comment:
those elements are so nested, isnt it possible to reduce this div tree?

Reply:
use predifined wrapping component with those css styles sience you have same approach in TablePage.